### PR TITLE
Fixed creation/import of groups and organizations when harvesting from remote ckan instance

### PR DESCRIPTION
**Pull request by polarp**
This is a fix for #240

### CODE REVIEW

File: ckanext/harvest/harvesters/ckanharvester.py [hunk 1]
```diff
@@ -123,10 +128,10 @@ def validate_config(self, config):
 
                 # Check if default groups exist
                 context = {'model': model, 'user': c.user}
-                for group_name in config_obj['default_groups']:
+                for group_ in config_obj['default_groups']:
```
Comment (amercader) [inline ckanext/harvest/harvesters/ckanharvester.py]: This seems wrong. `default_groups` is (or should be) a list of of strings (names or ids). Why do you assume they are a dict?

File: ckanext/harvest/harvesters/ckanharvester.py [hunk 2]
```diff
@@ -404,36 +409,30 @@ def import_stage(self, harvest_object):
                 # check if remote groups exist locally, otherwise remove
                 validated_groups = []
 
-                for group_name in package_dict['groups']:
+                for group_ in package_dict['groups']:
```
Comment (amercader) [inline ckanext/harvest/harvesters/ckanharvester.py]: This makes sense, I assume it was a remnant of the old API version 1/2 based harvester where `groups` is a list of strings.

File: ckanext/harvest/harvesters/ckanharvester.py (hunk 1 above)
Comment (polarp) [inline ckanext/harvest/harvesters/ckanharvester.py]: @amercader You're right, I overlooked this. I'll revert that change.

Comment (polarp) [inline ckanext/harvest/harvesters/ckanharvester.py]: @amercader Just to let you know that I reverted the change made in the default groups.

**Comment by amercader** [comment]
@polarp Thanks for the patch. I added a couple of tests for the remote group creation in 16a6e9fbf6eb53, you can use them as guidance if you need to send further patches.

**Pull request merged**